Initial synchronisation fails when areas are active but table is empty

<!--
If you've never submitted an issue to the SORMAS repository before or this is your first time using this template, please read the Contributing guidelines (https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) for an explanation of the information we need you to provide. You don't have to remove this comment or any other comment from this issue as they will automatically be hidden.
-->
### Bug Description
<!-- Mandatory -->
When initially syncing infrastructure data on a system that has the `INFRASTRUCTURE_TYPE_AREA` feature enabled, but no areas in the database, the synchronisation fails:

```
01-27 14:20:57.309 15087 22933 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
01-27 14:20:57.309 15087 22933 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AdoDtoHelper.lambda$handlePulledList$0(AdoDtoHelper.java:183)
01-27 14:20:57.309 15087 22933 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AdoDtoHelper.$r8$lambda$hCzJ3sszF_oP3vgBtfucOv2jnwg(Unknown Source:0)
01-27 14:20:57.309 15087 22933 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at de.symeda.sormas.app.backend.common.AdoDtoHelper$$ExternalSyntheticLambda0.call(Unknown Source:6)
01-27 14:20:57.309 15087 22933 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	at com.j256.ormlite.misc.TransactionManager.callInTransaction(TransactionManager.java:237)
01-27 14:20:57.309 15087 22933 E de.symeda.sormas.app.rest.SynchronizeDataAsync$1: 	... 15 more
```

### Steps to Reproduce
<!-- Optional; please add more steps if necessary -->
1. Set up a server with the feature configuration enabled, but don't create any area in the database
2. Install the mobile app and log in with a user to trigger the initial synchronization.

### Expected Behavior
<!-- Optional -->
The Android app should be able to handle empty or null lists here; check whether this problem might also appear for other infrastructure data (e.g. regions, districts, etc.) and if so, fix it there as well.

### Screenshots
<!-- Optional -->

### System Details
<!-- Mandatory; you only have to specify the Server URL if the error appeared on a publicly available test server -->
* Device:
* SORMAS version:
* Android version/Browser:
* Server URL:
* User Role:

### Additional Information
<!-- Optional -->

--- a/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/AdoDtoHelper.java
+++ b/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/AdoDtoHelper.java
@@ -177,6 +177,11 @@ public abstract class AdoDtoHelper<ADO extends AbstractDomainObject, DTO extends
 
 	public int handlePulledList(AbstractAdoDao<ADO> dao, List<DTO> result)
 		throws DaoException, NoConnectionException, ServerConnectionException, ServerCommunicationException {
+
+		if (result == null) {
+			return 0;
+		}
+
 		preparePulledResult(result);
 		dao.callBatchTasks((Callable<Void>) () -> {
 // 		boolean empty = dao.countOf() == 0;

--- a/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/AdoDtoHelper.java
+++ b/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/AdoDtoHelper.java
@@ -175,6 +175,10 @@ public abstract class AdoDtoHelper<ADO extends AbstractDomainObject, DTO extends
 		return 0;
 	}
 
+	/**
+	 * Calls handlePulledDto for each DTO that has been pulled from the server, and returns the
+	 * number of pulled DTOs.
+	 */
 	public int handlePulledList(AbstractAdoDao<ADO> dao, List<DTO> result)
 		throws DaoException, NoConnectionException, ServerConnectionException, ServerCommunicationException {
 

